Tests/FluentAssertions.Specs/Numeric/NumericAssertionSpecs.BeNaN.cs

```diff
@@ -0,0 +1,490 @@
+﻿using System;
+using Xunit;
+using Xunit.Sdk;
+
+namespace FluentAssertions.Specs.Numeric;
+
+public partial class NumericAssertionSpecs
+{
+    public class BeNaN
+    {
+        [Fact]
+        public void When_a_float_value_is_NaN_it_should_succeed()
```

Comment:
We have been moving away from the `When_[..]_it_should_[..]` naming scheme and over to a less strict naming that focuses more even more on the behavior being tested and less about fitting into to some scheme.
E.g. when most tests end with `it_should_succeed` that suffix provides little to no value.

See https://enterprisecraftsmanship.com/posts/you-naming-tests-wrong/